src/alert/src/InlineAlert.js

```diff
@@ -1,77 +1,69 @@
 import { Text } from '../../typography'
 import { Icon } from '../../icon'
 
-class InlineAlert extends PureComponent {
-  static propTypes = {
-    /**
-     * Composes some Box APIs.
-     */
-    ...spacing.propTypes,
-    ...position.propTypes,
-    ...layout.propTypes,
-    ...dimensions.propTypes,
+function InlineAlert({
+  children,
+  intent = 'none',
+  hasIcon = true,
+  size = 400,
+  ...props
+}, ref) {
+  const theme = useTheme()
 
-    /**
-     * The content of the alert.
-     */
-    children: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
-
-    /**
-     * The intent of the alert. This should always be set explicitly.
-     */
-    intent: PropTypes.oneOf(['none', 'success', 'warning', 'danger'])
-      .isRequired,
-
-    /**
-     * When true, show a icon on the left matching the type.
-     * There is no point not showing this.
-     */
-    hasIcon: PropTypes.bool,
-
-    /**
-     * The size of the Text.
-     */
-    size: PropTypes.number,
+  return (
+    <Pane alignItems="center" display="flex" innerRef={ref} {...props}>
+      {hasIcon && (
+        <Pane display="inline" marginRight={8}>
+          <Icon size={14} marginTop={2} {...theme.getIconForIntent(intent)} />
+        </Pane>
+      )}
+      <Text size={size} fontWeight={500}>
+        {children}
+      </Text>
+    </Pane>
+  )
+}
 
-    /**
-     * Theme provided by ThemeProvider.
-     */
-    theme: PropTypes.object.isRequired
-  }
+InlineAlert.propTypes = {
+  /**
+   * Composes some Box APIs.
+   */
+  ...spacing.propTypes,
+  ...position.propTypes,
+  ...layout.propTypes,
+  ...dimensions.propTypes,
 
-  static defaultProps = {
-    intent: 'none',
-    hasIcon: true,
-    size: 400
-  }
+  /**
+   * The content of the alert.
+   */
+  children: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
 
-  getIconForIntent = intent => {
-    const { theme } = this.props
+  /**
+   * The intent of the alert. This should always be set explicitly.
+   */
+  intent: PropTypes.oneOf(['none', 'success', 'warning', 'danger'])
+    .isRequired,
 
-    return <Icon size={14} marginTop={2} {...theme.getIconForIntent(intent)} />
-  }
+  /**
+   * When true, show a icon on the left matching the type.
+   * There is no point not showing this.
+   */
+  hasIcon: PropTypes.bool,
 
-  render() {
-    const { theme, children, intent, hasIcon, size, ...props } = this.props
+  /**
+   * The size of the Text.
+   */
+  size: PropTypes.number,
 
-    return (
-      <Pane alignItems="center" display="flex" {...props}>
-        {hasIcon && (
-          <Pane display="inline" marginRight={8}>
-            {this.getIconForIntent(intent)}
-          </Pane>
-        )}
-        <Text size={size} fontWeight={500}>
-          {children}
-        </Text>
-      </Pane>
-    )
-  }
+  /**
+   * Theme provided by ThemeProvider.
+   */
+  theme: PropTypes.object.isRequired
 }
 
-export default withTheme(InlineAlert)
+export default memo(forwardRef(InlineAlert))
```

Comment:
Do we need to set the `.displayName` of the component? Not sure if we need it but it might be nice.

Reply:
Not sure if what I did is what you're looking for, lmk if it is/isn't and I'll make those changes to the other PR's as well :)